packages-common/pluggable-widgets-tools/tests/commands.js

```diff
@@ -0,0 +1,175 @@
+
+    async function runTest(platform, lang, version) {
+        const widgetName = `generated_${version.replace(".", "_")}_${lang}_${platform}`;
+        const workDir = join(tempdir(), `pwt_test_${Math.round(Math.random() * 10000)}`);
+
+        try {
+            console.log(`[${widgetName}] Preparing widget...`);
+            await prepareWidget();
+            console.log(`[${widgetName}] Ready to test!`);
+
+            console.log(`[${widgetName}] Testing linting...`);
+            await testLint();
+            console.log(`[${widgetName}] Testing unit tests....`);
+            if (platform === "native") {
+                await testTest();
+            } else {
+                await execAsync("npm run test", workDir);
+            }
+
+            if (LIMIT_TESTS) {
+                console.log(`[${widgetName}] Quick tested!`);
+                return;
+            }
+
+            for (const cmd of ["build", "test:unit", "release"]) {
+                console.log(`[${widgetName}] Testing '${cmd}' command...`);
+                await execAsync(`npm run ${cmd}`, workDir);
+            }
+            console.log(`[${widgetName}] Testing npm start...`);
+            await testStart();
+            console.log(`[${widgetName}] Tested!`);
+        } finally {
+            await new Promise(resolve => setTimeout(resolve, 5000)); // time for all processes to die and unlock the folder
+            rm("-rf", workDir);
+        }
+
+        async function prepareWidget() {
+            mkdir(workDir);
+
+            if (version === "latest") {
+                let generatedWidget;
+                const release = await mutex.acquire(); // yeoman generator is no re-entrable :(
+                try {
+                    generatedWidget = await runYeoman(require.resolve("@mendix/generator-widget/generators/app"))
+                        .inTmpDir()
+                        .withPrompts({
+                            widgetName: "Generated",
+                            programmingLanguage: lang === "ts" ? "typescript" : "javascript",
+                            platform,
+                            e2eTests: false
+                        })
+                        .withArguments("Generated")
+                        .toPromise();
+                } finally {
+                    release();
+                }
+                await copy(join(generatedWidget, "generated"), workDir);
+            } else {
+                await copy(join(__dirname, "projects", widgetName), workDir);
+            }
+
+            const widgetPackageJson = await readJson(join(workDir, "package.json"));
+            widgetPackageJson.devDependencies["@mendix/pluggable-widgets-tools"] = toolsPackagePath;
+            await writeJson(join(workDir, "package.json"), widgetPackageJson);
+
+            await execAsync("npm install --loglevel=error", workDir);
+        }
+
+        async function testLint() {
+            await execFailedAsync("npm run lint", workDir);
```

Comment:
What makes lint fail when widget is generated?

Reply:
Files in our template is not pretty-formatted. Previously `npm run lint` would auto-format files, but i changed it in one of the previous PR (because it is unexpected for lint command to fix issues - we have `lint:fix` for that).